src/code/z_horse.c

```diff
@@ -93,190 +119,207 @@ void func_8006D0EC(PlayState* play, Player* player) {
+
+                    if (GET_EVENTINF_INGORACE_HORSETYPE() != HORSE_EPONA && horseSpawns[i].type == HORSE_PTYPE_6) {
+                        // HORSE_HNI
+                        paramFlag = 0x8000;
                     }
 
                     player->rideActor =
-                        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_HORSE, D_8011F9B8[i].pos.x, D_8011F9B8[i].pos.y,
-                                    D_8011F9B8[i].pos.z, 0, D_8011F9B8[i].angle, 0, D_8011F9B8[i].type | temp);
+                        Actor_Spawn(&play->actorCtx, play, ACTOR_EN_HORSE, horseSpawns[i].pos.x, horseSpawns[i].pos.y,
+                                    horseSpawns[i].pos.z, 0, horseSpawns[i].angle, 0, horseSpawns[i].type | paramFlag);
                     ASSERT(player->rideActor != NULL, "player->ride.actor != NULL", "../z_horse.c", 667);
 
-                    player->actor.world.pos.x = D_8011F9B8[i].pos.x;
-                    player->actor.world.pos.y = D_8011F9B8[i].pos.y;
-                    player->actor.world.pos.z = D_8011F9B8[i].pos.z;
+                    player->actor.world.pos.x = horseSpawns[i].pos.x;
+                    player->actor.world.pos.y = horseSpawns[i].pos.y;
+                    player->actor.world.pos.z = horseSpawns[i].pos.z;
                     player->actor.shape.rot.x = player->actor.shape.rot.z = 0;
-                    player->actor.shape.rot.y = D_8011F9B8[i].angle;
+                    player->actor.shape.rot.y = horseSpawns[i].angle;
 
                     Actor_MountHorse(play, player, player->rideActor);
-                    func_8002DE74(play, player);
+                    Actor_SetCameraHorseSetting(play, player);
 
-                    sp54.x = player->actor.world.pos.x - 200.0f;
-                    sp54.y = player->actor.world.pos.y + 100.0f;
-                    sp54.z = player->actor.world.pos.z;
+                    eye.x = player->actor.world.pos.x - 200.0f;
+                    eye.y = player->actor.world.pos.y + 100.0f;
+                    eye.z = player->actor.world.pos.z;
 
-                    Play_SetCameraAtEye(play, play->activeCamId, &player->actor.world.pos, &sp54);
+                    Play_SetCameraAtEye(play, play->activeCamId, &player->actor.world.pos, &eye);
                 } else {
-                    Actor_Spawn(&play->actorCtx, play, ACTOR_EN_HORSE, D_8011F9B8[i].pos.x, D_8011F9B8[i].pos.y,
-                                D_8011F9B8[i].pos.z, 0, D_8011F9B8[i].angle, 0, D_8011F9B8[i].type);
+                    Actor_Spawn(&play->actorCtx, play, ACTOR_EN_HORSE, horseSpawns[i].pos.x, horseSpawns[i].pos.y,
+                                horseSpawns[i].pos.z, 0, horseSpawns[i].angle, 0, horseSpawns[i].type);
                 }
                 break;
             }
         }
     }
 }
 
-void func_8006DC68(PlayState* play, Player* player) {
+/**
+ * Determines when and where a horse should spawn for the player
+ */
+void Horse_Set(PlayState* play, Player* player) {
```

Comment:
I'm against calling it Actor_SpawnHorse and somewhat against using the word Spawn at all because it doesn't fit the same sort of operation that Actor_Spawn does. With Actor_Spawn, there is a reasonable expectation that the function will always return a new instance to an actor, whereas "`Horse_Set`" will only create a horse if the rather complex set of conditions to do so are met.

Reply:
I think `Set` does a worse job of capturing "create a horse if the rather complex set of conditions to do so are met" though, compared to `Spawn`. Maybe something like `Horse_TrySpawn`?